target/bin/sedfile

```diff
@@ -11,36 +11,35 @@
 
 set -ueo pipefail
 
-HASHTOOL="sha1sum"
-SKIP_ERROR=0
+HASHTOOL='sha1sum'
+SKIP_ERROR=false
 
 if [[ $# -lt 3 ]]
 then
-  echo "Error:  At least, three parameters must be given."
-  echo "Syntax: sedfile -i <replace/delete operation> <file>"
-  echo
+  echo 'At least three parameters must be given'
+  echo -e "Usage: ${0} -i <replace/delete operation> <file>\n"
   exit 1
 fi >&2
 
-[[ -f /CONTAINER_START ]] && SKIP_ERROR=1 # Hide error, if container was restarted.
-if [[ "${1}" == "--strict" ]]         # Show error every time.
+[[ -f /CONTAINER_START ]] && SKIP_ERROR=true # hide error if container was restarted
+if [[ ${1} == '--strict' ]]                  # show error every time
 then
-  SKIP_ERROR=0
+  SKIP_ERROR=false
   shift
 fi
 
 # get last argument
 FILE=${*: -1}
 
-OLD=$(${HASHTOOL} "${FILE}")
-sed "$@"
-NEW=$(${HASHTOOL} "${FILE}")
+OLD="$(${HASHTOOL} "${FILE}")"
+sed "${@}"
+NEW="$(${HASHTOOL} "${FILE}")"
 
 # fail if file was not modified
-if [[ ${OLD} == "${NEW}" ]] && [[ ${SKIP_ERROR} -eq 0 ]]
+if [[ ${OLD} == "${NEW}" ]] && ! ${SKIP_ERROR}
 then
-  echo "Error: sed $*"
+  echo "Error: sed ${*}" >&2
   exit 1
-fi >&2
+fi
```

Comment:
Any reason this echo isn't using log like others? I'm not quite sure what the decision process is for when to convert.

The if statements that end with `fi >&2` seem odd to me, so I like that you addressed that here, it's present elsewhere in files touched by this PR if that is of interest.

Reply:
This `if` statement only logs one message, so I thought it is fine to place the redirection right behind it. The other statement above logs more than one, all going to `stderr`, so I left it after the `fi`.

Reply:
We should use `_log` here instead of `echo`?

Reply:
We can't / shouldn't (for the reasons described down below) :)